<!-- Launch file for the rrt-detector and the assigner -->

<launch>
<arg name="eta" value="1.0"/>
<arg name="Geta" value="1.0"/>
<param name="namespace_init_count" value="1"/>


  <node pkg="turtlebot3_aslam" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="/map"/>
  <param name="costmap_topic" value="/move_base/global_costmap/costmap"/>
  <param name="robot_frame" value="/base_link"/>
  </node>
  
  <node pkg="turtlebot3_aslam" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="/map"/>
  <param name="costmap_topic" value="/move_base/local_costmap/costmap"/>
  <param name="robot_frame" value="/base_link"/>
  </node>
 
  
  <node pkg="turtlebot3_aslam" type="filter.py" name="filter" output="screen">
  <param name="map_topic" value="/map"/>
  <param name="door_length" value="2.0"/>
  <param name="wall_thickness" value="0.2"/>
  <param name="robot_length" value="0.1"/>
  <param name="info_radius" value="1.0"/>
  <param name="costmap_clearing_threshold" value="90"/>
  <param name="global_costmap_topic" value="/move_base/global_costmap/costmap"/>
  <param name="local_costmap_topic" value="/move_base/local_costmap/costmap"/>
  <param name="goals_topic" value="/detected_points"/>
  <param name="rate" value="100"/>
  </node>
  
  <node pkg="turtlebot3_aslam" type="assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="/map"/>
  <param name="global_frame" value="/map"/>
  <param name="info_radius" value="1"/> 
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="delay_after_assignement" value="1.0"/>
  <param name="rate" value="100"/>
  </node>
  
</launch>
